packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -134,6 +145,16 @@ export const addItemToCart = (payload = {}) => {
     };
 };
 
+// Returns true if the cart is invalid.
+function hasRetryableGqlError(error) {
```

Comment:
This function feels nasty. I don't like matching some string value from the API. Would rather we be able to use some error id map. GQL always returns 200 so we no longer get 404 for invalid carts.

Reply:
Turns out that `category: "graphql-no-such-entity"` is used if anything within the response is not found. To ensure we only retry on an invalid cart, and not on an invalid/oos item, we should use the `message` string.